src/roslib.c

```diff
@@ -0,0 +1,34 @@
+#include <lua.h>
+#include <lauxlib.h>
+#include <lualib.h>
+#include <locale.h>
+
+/*
+Enable setlocale to eliminate differences in the results 
+of the sort function in different regions. Note that the 
+category is restricted to be set to 'collate' only cuz we
+only concern about the result of strcoll() function.
+*/
+static int r_os_setlocale(lua_State *L){
+    int argc = lua_gettop(L);
+    char *s;
+
+    if (argc != 1) {
+        lua_pushstring(L, "wrong number of arguments");
+        return lua_error(L);
+    }
+    const char *l = luaL_optstring(L, 1, NULL);
+    lua_pushstring(L, setlocale(LC_COLLATE, l));
```

Comment:
I don't think we wanna change the system (global) locale in a script, it'll have implications outside that script.
maybe instead we should expose a CONFIG directive to control it?
i.e. currently redis relies on environment variable for that.
see https://github.com/redis/redis/issues/1074 and https://github.com/redis/redis/pull/799

Reply:
CONFIG is indeed better cuz the environment is shared. I'll make changes in later commits.

Reply:
I think controlling locale through environment is more standard, having a configuration parameter that overrides that might be very counter intuitive in some cases. If a Lua script can change the locale programmatically, and that change only affects that script - I think that's much better.

Reply:
i'm not sure it's at all possible to have it only affect Lua.
indeed it's standard to control the locale via env var, but that's actually not very convenient.
i think that (despite being non-standard), it's much more useful to set such things via config file and config command).
same as we handle other global server configs like `rdbcompression` and ziplist threshold.
although granted, these do not affect behavior, just optimizations.

Reply:
Since it seems necessary to uniform the output format of SORT command globally (eg. the same business deployed in different regions), it may be better to control it from the upper level, i.e. CONFIG command.

Reply:
The title of this pr is  easily misunderstood and  I've created a new pr #11059